### DISCUSSION TITLE
The future of V1 and V2

### DISCUSSION THREAD

**Discussion by kpym** [tectonic-typesetting/tectonic#993] [Ideas]
I would like to present my very personal vision of the future interface of `tectonic`. My main idea is that we do not have to choose between V1 and V2. They can co-exist in a very natural way (without the use of extra `-X` or `-Y`).

The V1 could be called "file interface" and is activated when the target is a file. Like in `tectonic main.tex`. This will be equivalent to `tectonic compile main.tex`.

The V2 could be called "project interface" and is enabled when the target is a folder, as in `tectonic .` (to build the project in the current folder, which will be the same as `tectonic build .`). All other explicit subcommands are unambiguous, so we can drop the `-X`.

My point is that it is important to keep the V1 interface as the default for files. This way `tectonic` could be seen as a member of the latex family (`xetex`, `pdflatex`...). And dropping `-X` to make V2 first class citizen can be done without forcing `-Y` for file compilation.

**Comment by pkgw** [comment]
I've realized that the `-Y` idea is now unnecessary even without adding any extra functionality — it can just be subsumed into `tectonic compile ...`. That part of the documentation should be updated to reflect that.

Overall my feeling is that it is better to have Tectonic behave like `git` and `cargo` and other tools and *always* require a subcommand. Besides some corner-case possibilities for confusion (what if you have an input file named `watch`?), I don't want to privilege the pass-a-file-on-the-command-line mode because, as per some other recent discussions here and in the issues, that mode inevitably raises some issues with regards to reproducible and reliable builds that can (at least in principle) be avoided in the V2 "document model" approach with a `Tectonic.toml` and, one day, `Tectonic.lock` file.
